hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineManagerImpl.java

```diff
@@ -697,6 +698,28 @@ public Map<String, Integer> getPipelineInfo() throws NotLeaderException {
     return pipelineInfo;
   }
 
+  @Override
+  public Map<String, String[]> getPipelineLeaders() throws NotLeaderException {
+    final Map<String, String[]> pipelineInfo = new HashMap<>();
+    List<Pipeline> pipelines = getPipelines();
+    pipelines.forEach(pipeline -> {
+      String leaderNode = "";
+      UUID pipelineId = pipeline.getId().getId();
+
+      try {
+        leaderNode = pipeline.getLeaderNode().getHostName();
```

Comment:
I know that for users the datanode hostname would be more informative, but as I looked into it in the [DatanodeDetails](https://github.com/apache/ozone/blob/c7038e08a8df296108280f3759be55e7cbbf1d78/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/protocol/DatanodeDetails.java) class only the UUID is unique, based on this [code snippet](https://github.com/apache/ozone/blob/c7038e08a8df296108280f3759be55e7cbbf1d78/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/protocol/DatanodeDetails.java#L509L523). Maybe we should use that here or add both?

Reply:
I followed a similar approach to how recon display's the Leader DataNode of the pipeline, by displaying the HostName. 
<img width="571" alt="image" src="https://user-images.githubusercontent.com/98023601/207059287-da0723ec-7b43-4d7d-8cc1-1705cf891584.png">

Reply:
Hi @ArafatKhan2198 , I think we should show Datanode UUID as well, since we might want to run hundreds of (mock) Ozone datanodes in the future for stress testing. Recon would need to be patched accordingly too.

Reply:
Yes its probably a good idea then, I have added it to the metrics!
@smengcl